### ISSUE TITLE
darkreading.com

### ISSUE THREAD

**Issue by SeriousHoax** [uBlockOrigin/uAssets#13938]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

https://www.darkreading.com/

### Describe the issue

When I visit `https://www.darkreading.com/` the websites comes into display with a 10 seconds delay. Clicking on any article on the website results in a similar issue. The only way to fix this is to disable "uBlock filters".

### uBlock Origin version

1.43.0

### Browser name and version

Firefox 102.0

### Settings

Filter lists: Default uBO, EasyList, EasyPrivacy, Anti-Facebook, Adguard Annoyance

**Comment by SeriousHoax** [comment]
@okiehsch The fix doesn't work when Firefox's built-in tracking protection is enabled. With Adguard extension, the issue doesn't occur even when Firefox's tracking protection is on. Can you do something about it in uBO?

**Comment by okiehsch** [comment]
> The fix doesn't work when Firefox's built-in tracking protection is enabled. 

The issue occurs even if you disable uBO with that setup, so there is nothing to fix for me as far as I can see.
I don't know why it works with Adguard, they don't have any specific filters for that site.

**Comment by SeriousHoax** [comment]
> > The fix doesn't work when Firefox's built-in tracking protection is enabled.
> 
> The issue occurs even if you disable uBO with that setup, so there is nothing to fix for me as far as I can see. I don't know why it works with Adguard, they don't have any specific filters for that site.

Yeah, I also saw that. It's weird that it's working with Adguard even though don't see anything noteworthy in logs. Alright then, no problem.

**Comment by Yuki2718** [comment]
On my end enabling ETP is not enough but setting it to Strict is required (Standard allows doubleclick). The cause is that uBF redirects `https://beta.darkreading.com/js/prebid-ads/adsensebase.js` to avoid detection while AG doesn't. Here's the culprit taken from `https://beta.darkreading.com/_next/static/chunks/pages/_app-ed3bf26a215597ef97e2.js`
```
(this, "unhiddenMainContent", (function() {
                            var e = document.getElementById(s.N.mainContentId),
                                t = document.getElementById(s.N.overlayFixedAllScreenId);
                            e && e.style.removeProperty("opacity"), t && t.classList.add("d-none")
                        })), (0, o.Z)(this, "checkAdsBlock", (function() {
                            return !window.canRunAds ? t.unhiddenMainContent("isBlocked") : t.unhiddentTimeout = setTimeout((function() {
                                return t.unhiddenMainContent("end timeout")
                            }), 1e4)
```
They set literally 10s dealy if not detected. Simple fix is stop redirecting i.e.
```
/prebid-ads/adsensebase.js$script,redirect-rule=prebid-ads.js,domain=~darkreading.com
/prebid-ads/adsensebase.js$script,redirect-rule=prebid-ads.js,badfilter
```
An alternative to keep detection avoided is 
```
darkreading.com###overlay-fixed-all-screen
darkreading.com###container-main:style(opacity: 1 !important;)
```
(or `darkreading.com##+js(nano-stb, unhiddenMainContent, 10000, 0.02)`) IDK which is better in this case.

**Comment by okiehsch** [comment]
> ETP is not enough but setting it to Strict is required (Standard allows doubleclick).

@SeriousHoax we don't add filters for issues with customised setups, if the suggested filters work just add them to your personal filter list.

**Comment by Yuki2718** [comment]
@okiehsch But this is *in a sense* caused by our filter `/prebid-ads/adsensebase.js$script,redirect-rule=prebid-ads.js`.

**Comment by okiehsch** [comment]
Ok, I can replace my fix from yesterday with 
`@@/prebid-ads/adsensebase.js$script,redirect-rule=prebid-ads.js,domain=darkreading.com`
if that fixes the issue with Firefox tracking protection too.

**Comment by okiehsch** [comment]
@Yuki2718 I can reproduce the issue even if I disable uBO with Firefox as I mentioned earlier, so I am not sure how "we" cause the issue.

**Comment by Yuki2718** [comment]
That's because ETP (-strict on my side) blocks doubleclick but not `/adsensebase.js`. To not see the delay, it's either allow everything or block both (without redirect).

**Comment by okiehsch** [comment]
So, it's Firefox "fault" not ours imo.
Still because your filter works with all browsers - and a specific setup - it's better than mine.

**Comment by Yuki2718** [comment]
If not site's fault - I have no idea why they made it like this.

**Comment by Yuki2718** [comment]
Oh, sorry it seems if ETP is off or Standard `@@||securepubads.g.doubleclick.net/tag/js/gpt.js$script,redirect-rule=googletagservices_gpt.js:5,domain=darkreading.com` is still needed.
